test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -796,4 +1225,270 @@ class QueryTests {
+                returnType.isMarkedNullable -> NULLABLE_DOUBLE_VALUES.mapNotNull { it }.maxOrNull()
+                else -> DOUBLE_VALUES.maxOrNull()
+            }
+            AggregatorQueryType.SUM -> when {
+                returnType.isMarkedNullable -> NULLABLE_DOUBLE_VALUES.sumOf { it ?: 0.0 }
+                else -> DOUBLE_VALUES.sum()
+            }
+        }?.toDouble() ?: throw IllegalArgumentException("Aggregate result cannot be null.")
+        Float::class -> when (type) {
+            AggregatorQueryType.MIN -> when {
+                returnType.isMarkedNullable -> NULLABLE_FLOAT_VALUES.mapNotNull { it }.minOrNull()
+                else -> FLOAT_VALUES.minOrNull()
+            }
+            AggregatorQueryType.MAX -> when {
+                returnType.isMarkedNullable -> NULLABLE_FLOAT_VALUES.mapNotNull { it }.maxOrNull()
+                else -> FLOAT_VALUES.maxOrNull()
+            }
+            AggregatorQueryType.SUM -> when {
+                returnType.isMarkedNullable -> NULLABLE_FLOAT_VALUES.sumOf { it?.toInt() ?: 0 }
+                else -> FLOAT_VALUES.sum()
+            }
+        }?.toFloat() ?: throw IllegalArgumentException("Aggregate result cannot be null.")
+        RealmInstant::class -> when (type) {
+            AggregatorQueryType.MIN -> when {
+                returnType.isMarkedNullable ->
+                    NULLABLE_TIMESTAMP_VALUES.mapNotNull { it }.minOrNull()
+                else -> TIMESTAMP_VALUES.minOrNull()
+            }
+            AggregatorQueryType.MAX -> when {
+                returnType.isMarkedNullable ->
+                    NULLABLE_TIMESTAMP_VALUES.mapNotNull { it }.maxOrNull()
+                else -> TIMESTAMP_VALUES.maxOrNull()
+            }
+            AggregatorQueryType.SUM -> throw IllegalArgumentException("SUM is not allowed on timestamp fields.")
+        } ?: throw IllegalArgumentException("Aggregate result cannot be null.")
+        else -> throw IllegalArgumentException("Only numerical properties and timestamps are allowed.")
+    }
+
+    // ------------------------
+    // General-purpose helpers
+    // ------------------------
+
+    // Deletes all objects after assertions to avoid "null vs 0" results when testing aggregators
+    private fun cleanUpBetweenProperties() = realm.writeBlocking {
+        query(QuerySample::class)
+            .find { results ->
+                results.toList() // We cannot iterate over the results and delete directly!
+                    .forEach { sample ->
+                        val latest = findLatest(sample)
+                        assertNotNull(latest)
+                        delete(latest)
+                    }
+            }
+    }
+
+    // ----------------------------------------
+    // Descriptors used for exhaustive testing
+    // ----------------------------------------
+
+    private val basePropertyDescriptors = listOf(
+        PropertyDescriptor(QuerySample::intField, Int::class, INT_VALUES),
```

Comment:
Seems weird that we test `Short` explicitly. All integral types map to Core's INTEGER, so if we test all Kotlin integral types shouldn't we then also test `Char` and `Byte`?